azkaban-common/src/main/java/azkaban/project/JdbcProjectLoader.java

```diff
@@ -385,6 +385,22 @@ private void uploadProjectFile(Connection connection, int projectId, int version
     updateChunksInProjectVersions(connection, projectId, version, chunks);
   }
 
+  public void addProjectVersion(
+      int projectId,
+      int version,
+      File localFile,
+      String uploader,
+      String uri) throws ProjectManagerException {
+    try (Connection connection = getConnection()) {
+      addProjectToProjectVersions(connection, projectId, version, localFile, uploader, uri);
+      connection.commit();
+    } catch (SQLException e) {
+      logger.error(e);
+      throw new ProjectManagerException(String.format("Add ProjectVersion failed. project id: %d version: %d",
+          projectId, version), e);
+    }
+  }
```

Comment:
@kunkun-tang 
I think `close()` will be automatically called on connection since it is being called in a [try-with-resources](https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html) mechanism. Let me know if you still think it's a problem